qemu/tests/qemu_guest_agent.py

```diff
@@ -1554,6 +1555,125 @@ def setup(self, test, params, env):
+
+        @error_context.context_aware
+        def result_check(flag, write_timeout):
+            """
+            Check if freeze or thaw guest file system works.
+
+            :param flag: frozen or thaw
+            :param write_timeout: timeout of writing to guest file
+            """
+            if self.fswrite_thread:
+                error_context.context("Waiting %s, then finish writing the time "
+                                      "stamp in guest file." % write_timeout)
+                self.fswrite_thread.join(write_timeout)
+                if self.fswrite_thread.isAlive():
+                    session2 = self._get_session(self.params, None)
+                    self._open_session_list.append(session2)
+                    k_cmd = "wmic process where \"name='python.exe' and " \
+                            "CommandLine Like '%fsfreeze%'\" call terminate"
+                    s, o = session2.cmd_status_output(k_cmd)
+                    if s:
+                        self.test.error("Command '%s' failed, status: %s, "
+                                        "output: %s" % (k_cmd, s, o))
+            error_context.context("Check guest FS status.", logging.info)
+            # init fs status to 'thaw'
+            fs_status = "thaw"
+            file_name = "/tmp/fsfreeze_%s.txt" % flag
+            process.system("rm -rf %s" % file_name)
+            self.vm.copy_files_from("C:\\fsfreeze.txt", file_name)
+            with open(file_name, 'r') as f:
+                list_time = f.readlines()
+
+            for i in list(range(0, len(list_time))):
+                list_time[i] = list_time[i].strip()
+
+            for i in list(range(1, len(list_time))):
+                num_d = float(list_time[i]) - float(list_time[i - 1])
+                if num_d > 8:
+                    logging.info("Time stamp is not continuous,"
+                                 " so the FS is frozen.")
+                    fs_status = "frozen"
+                    break
+            if not fs_status == flag:
+                self.test.fail("FS is not %s, it's %s." % (flag, fs_status))
+
+        error_context.context("Check guest agent command "
+                              "'guest-fsfreeze-freeze/thaw'",
+                              logging.info)
+        # make write time longer than freeze timeout
+        write_timeout = int(params["freeze_timeout"]) + 10
+        try:
+            expect_status = self.gagent.FSFREEZE_STATUS_THAWED
+            self.gagent.verify_fsfreeze_status(expect_status)
+        except guest_agent.VAgentFreezeStatusError:
+            # Thaw guest FS if the fs status is incorrect.
+            self.gagent.fsthaw(check_status=False)
+
+        background_start()
+        error_context.context("Freeze the FS.", logging.info)
+        self.gagent.fsfreeze()
+        try:
+            result_check("frozen", write_timeout)
+            # Next, thaw guest fs.
+            background_start()
```

Comment:
It is not a good idea to use `InterruptedThread` here, cause the exception of sub thread might not be caught by the main thread as we expected.
Another issue is, you assign `self.fswrite_thread` to another thread started in the second  `background_start()`, so later on what you `join()` is the second one, is that what you want?

Reply:
well, I will have & to run it as a command in background